newIDE/app/src/Utils/GDevelopServices/User.js

```diff
@@ -48,3 +75,68 @@ export const getUserPublicProfile = (
     .get(`${GDevelopUserApi.baseUrl}/user-public-profile/${id}`)
     .then(response => response.data);
 };
+
+const postBadgeIfLocked = (
+  authenticatedUser: AuthenticatedUser,
+  achievementId: string
+): ?Promise<?Badge> => {
```

Comment:
Returning *maybe* a promise (containing *maybe* a badge) seems weird/dangerous.

Instead, always return a promise:
```suggestion
): Promise<?Badge> => {
```

So that the caller can safely `await` you. You can then either:
* mark the function `async`, and benefit from await and from the automatic boxing of `return null` into a `return Promise.resolve(null)`.
* or just replace `return null` by `return Promise.resolve(null)`.

I would go the async route if possible, but the second option work.
The important is to always return a promise, even if it turns out to be immediately set to a result.

Reply:
I knew that when writing this, there was something odd 😓